### PR TITLE
Optimize rc patch by acquiring envoyfilterwrappr once per node

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/httproute.go [hunk 1]
```diff
@@ -47,13 +47,15 @@ func (configgen *ConfigGeneratorImpl) BuildHTTPRoutes(node *model.Proxy, push *m
 	routeNames []string) []*route.RouteConfiguration {
 	routeConfigurations := make([]*route.RouteConfiguration, 0)
 
+	efw := push.EnvoyFilters(node)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/httproute.go:50]: only get the envoyfilterwrapper once per node rather than per route

**Review by ramaraochavali — APPROVED** [review]
Nice improvement !!!

**Review by howardjohn — COMMENTED** [review]
lgtm

**Pull request merged**